wagtail/wagtailcore/models.py

```diff
@@ -391,6 +391,15 @@ def _get_autogenerated_slug(self, base_slug):
         candidate_slug = base_slug
         suffix = 1
         parent_page = self.get_parent()
+        slug_parts = base_slug.split('-')
+
+        try:
+            slug_parts[-1] = str(int(slug_parts[-1]) + 1)
+            candidate_slug = '-'.join(slug_parts)
+        except ValueError:
+            # If the last part of the slug fails to be converted
+            # to an int then simply move on to auto incrementing
+            pass
 
         while not Page._slug_is_available(candidate_slug, parent_page, self):
             # try with incrementing suffix until we find a slug which is available
```

Comment:
Ok so I understand the scenario you have described above, but I'm not sure if there is a way to tell the intention behind what the user entered. Would it make more sense to have this setting applied on the model instead of a global setting?

The root problem is that for certain page types there may be a high chance of name collisions. Perhaps there's a different solution. Ideally for something like an events calendar I would normally construct a url like some-event-name-here-1234 or /1243/some-event-name or something where the ID does the lookup and the slug is actually just there for SEO purposes. In addition when the slug does not match a 301 redirect is performed to the correct version so if a user just entered something like hello-wrong-name-1234 a redirect would take them to some-event-name-1234.

Ultimately I'm trying to provide a smooth user experience by reducing the number of unnecessary form errors. But I do see your point that there is a scenario where a number might be intentional and shouldn't be incremented.

I've thought about using a snippet instead, however then you lose the ability to have the wagtail userbar to edit the snippets easily as well as panels and potentially everything else that goes with a page.